include/nuttx/mm/map.h

```diff
@@ -43,7 +49,11 @@ struct mm_map_entry_s
   off_t offset;
   int prot;
   int flags;
-  FAR void *priv;
+  union
+  {
+    FAR void *p;
+    int i;
```

Comment:
we can, but shouldn't do.

- for uintptr_t it is ok, but uintptr_t is not C89/90  (and is optional even in newer)
- direct cast between int and void * is illegal in principle (different size, some compilers even warn about this: "cast from pointer to integer of different size")
- it is much cleaner to type priv.p = ptr; and priv.i = i; than adding typecasts
- using union is extendable to other types without breaking any existing functionality
- most importantly, I like it like this ;)

Reply:
> we can, but shouldn't do.
> 
> * for uintptr_t it is ok, but uintptr_t is not C89/90  (and is optional even in newer)

C89 require is for toolchain which means we can't use the compiler specific new feature. But since libc is provided by NuttX self, we can use C99 or even newer library feature in common code if NuttX's libc provide the support. 

> * direct cast between int and void * is illegal in principle (different size, some compilers even warn about this: "cast from pointer to integer of different size")
> * it is much cleaner to type priv.p = ptr; and priv.i = i; than adding typecasts
> * using union is extendable to other types without breaking any existing functionality
> * most importantly, I like it like this ;)

if you want uintptr_t is better than int.

Reply:
I like the enum way, do you see it unacceptable?

Reply:
@jlaitine @xiaoxiang781216 good point about uintptr_t, note that it is widespread in many places inside nuttx/drivers/, nuttx/wireless/ and even inside nuttx/tools/. Since we are not testing it on any board that really requires C89, what you think if we force the NuttX Simulator to use C89?

Reply:
I think it is, in general, a good idea to include some target which forces using C89 into CI.

I am myself not that pedantic about that, as long as all the supported boards compile with the respective compilers, we should be good.

Reply:
I measured the difference to have it always included on nucleo-f091rc:nsh

With mapping code always included the size is 40612 bytes, and without it is  40492 bytes. The difference is just 120 bytes in code, which originates from fs_mumap.c.

So I think it is ok to remove the flag and have it always included. Will do that